src/Aspire.Hosting/Dashboard/DashboardServiceHost.cs

```diff
@@ -1,29 +1,41 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Collections.Immutable;
 using System.Net;
 using Aspire.Hosting.ApplicationModel;
 using Aspire.Hosting.Dcp;
+using Aspire.Hosting.Publishing;
+using Aspire.Hosting.Utils;
 using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Hosting.Server;
+using Microsoft.AspNetCore.Hosting.Server.Features;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
 using Microsoft.AspNetCore.Server.Kestrel.Core;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
-using Aspire.Hosting.Publishing;
-using Aspire.Hosting.Utils;
 
 namespace Aspire.Hosting.Dashboard;
 
 /// <summary>
-/// Hosts a gRPC service (via <see cref="DashboardService"/>) that a dashboard can connect to.
+/// Hosts a gRPC service via <see cref="DashboardService"/> (aka the "Resource Service") that a dashboard can connect to.
 /// Configures DI and networking options for the service.
 /// </summary>
 internal sealed class DashboardServiceHost : IHostedService
 {
+    /// <summary>
+    /// Name of the environment variable that optionally specifies the resource service URL,
+    /// which the dashboard will connect to over gRPC.
+    /// </summary>
+    /// <remarks>
+    /// This is primarily intended for cases outside of the local developer environment.
+    /// If no value exists for this variable, a port is assigned dynamically.
+    /// </remarks>
     private const string DashboardServiceUrlVariableName = "DOTNET_DASHBOARD_GRPC_ENDPOINT_URL";
-    private const string DashboardServiceUrlDefaultValue = "http://localhost:18999";
```

Comment:
This default is removed. If no value exists in the `DOTNET_DASHBOARD_GRPC_ENDPOINT_URL` environment of the app host, then a dynamic port is bound and exposed via `GetUrisAsync`.